skrub/datasets/tests/test_utils.py

```diff
@@ -1,18 +1,54 @@
+import shutil
+import tempfile
 from pathlib import Path
-from unittest import mock
 
+from skrub.datasets._utils import get_data_dir, get_data_home
 
-@mock.patch("os.path.dirname")
-def test_get_data_dir(mock_os_path_dirname):
+
+def test_get_data_dir():
```

Comment:
```suggestion
@pytest.mark.parametrize("data_home_type", ["string", "path"])
def test_get_data_dir_existing_folder(data_home_type):
```

Reply:
We can parametrize with `subdirectory_name` and it will avoid repeating code below